link/src/main/java/com/meshconnect/link/ui/LinkContract.kt

```diff
@@ -0,0 +1,46 @@
+package com.meshconnect.link.ui
+
+import android.content.Context
+import android.content.Intent
+import androidx.activity.result.contract.ActivityResultContract
+
+/**
+ * Implements [ActivityResultContract] to start catalog activity
+ * and returns [LinkResult] in callback.
+ * ```
+ *  private val linkLauncher = registerForActivityResult(LinkContract()) { result ->
+ *      when(result) {
+ *          is LinkResult.Success -> {
+ *              // success transaction
+ *              Log.d("LinkResult", "Success. ${result.payloads}")
+ *          }
+ *          is LinkResult.Exited -> {
+ *              // user exited the flow by clicking on back or close button
+ *              // probably because of an error
+ *              Log.d("LinkResult", "Exited. ${result.errorMessage}")
+ *          }
+ *      }
+ *  }
+ *
+ *  linkButton.setOnClickListener {
+ *      linkLauncher.launch("linkToken")
+ *  }
+ * ```
+ */
```

Comment:
do we need commented code ?

Reply:
This code snippet is used for documentation. You can run `./gradlew link:dokkaHtml` to generate it.
https://imgur.com/a/D4aB3yg

Btw, this is a good catch, I forgot to update some lines. Thx.